cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1695,6 +1695,15 @@ private AppFabricServiceStatus stop(Id.Program identifier, ProgramType type) {
       controller.stop().get();
       return AppFabricServiceStatus.OK;
     } catch (Throwable throwable) {
+      if (type == ProgramType.WORKFLOW) {
```

Comment:
I don't understand this logic. How the workflow is different than other program type? Also, does it behave the same in standalone vs distributed mode?